src/app/core/services/modal.service.ts

```diff
@@ -0,0 +1,20 @@
+import { Injectable } from '@angular/core';
+import { BehaviorSubject } from 'rxjs';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class ModalService {
+  isVisible = new BehaviorSubject(false);
+  isVisible$ = this.isVisible.asObservable();
```

Comment:
there is a common patttern with a getter to prevent a second instance variable and to create that observable lazy (on the first request)

```
get isVisible$ {
  return  this.isVisible.asObservable();
}


same same, and actually more code